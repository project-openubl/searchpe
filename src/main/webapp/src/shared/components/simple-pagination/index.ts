export { SimplePagination } from "./simple-pagination";
