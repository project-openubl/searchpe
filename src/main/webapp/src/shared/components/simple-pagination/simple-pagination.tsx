import React from "react";
import { Pagination, PaginationVariant } from "@patternfly/react-core";

export interface SimplePaginationProps {
  count: number;
  params: {
    perPage?: number;
    page?: number;
  };

  isTop?: boolean;
  isCompact?: boolean;
  perPageOptions?: number[];
  onChange: ({ page, perPage }: { page: number; perPage: number }) => void;
}

export const SimplePagination: React.FC<SimplePaginationProps> = ({
  count,
  params,
  isTop,
  isCompact,
  perPageOptions,
  onChange,
}) => {
  const mapPerPageOptions = (options: number[]) => {
    return options.map((option) => ({
      title: String(option),
      value: option,
    }));
  };

  const getPerPage = () => {
    return params.perPage || 10;
  };

  return (
    <Pagination
      itemCount={count}
      page={params.page || 1}
      perPage={getPerPage()}
      onPageInput={(_, page) => {
        onChange({ page, perPage: getPerPage() });
      }}
      onSetPage={(_, page) => {
        onChange({ page, perPage: getPerPage() });
      }}
      onPerPageSelect={(_, perPage) => {
        onChange({ page: 1, perPage });
      }}
      isCompact={isTop || isCompact}
      widgetId="pagination-options-menu"
      variant={isTop ? PaginationVariant.top : PaginationVariant.bottom}
      perPageOptions={mapPerPageOptions(perPageOptions || [10, 20, 50, 100])}
    />
  );
};
