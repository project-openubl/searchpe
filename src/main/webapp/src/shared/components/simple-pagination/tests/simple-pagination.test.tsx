import React from "react";
import { shallow } from "enzyme";
import { SimplePagination } from "../simple-pagination";

describe("SimplePagination", () => {
  it("Renders without crashing", () => {
    const wrapper = shallow(
      <SimplePagination
        count={15}
        params={{ page: 1, perPage: 10 }}
        onChange={jest.fn}
      />
    );
    expect(wrapper).toMatchSnapshot();
  });
});
