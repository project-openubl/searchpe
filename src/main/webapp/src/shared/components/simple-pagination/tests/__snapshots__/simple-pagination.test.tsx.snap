// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SimplePagination Renders without crashing 1`] = `
<Pagination
  className=""
  defaultToFullPage={false}
  firstPage={1}
  isCompact={false}
  isDisabled={false}
  isSticky={false}
  itemCount={15}
  itemsEnd={null}
  itemsStart={null}
  offset={0}
  onFirstClick={[Function]}
  onLastClick={[Function]}
  onNextClick={[Function]}
  onPageInput={[Function]}
  onPerPageSelect={[Function]}
  onPreviousClick={[Function]}
  onSetPage={[Function]}
  ouiaSafe={true}
  page={1}
  perPage={10}
  perPageOptions={
    Array [
      Object {
        "title": "10",
        "value": 10,
      },
      Object {
        "title": "20",
        "value": 20,
      },
      Object {
        "title": "50",
        "value": 50,
      },
      Object {
        "title": "100",
        "value": 100,
      },
    ]
  }
  titles={
    Object {
      "currPage": "Current page",
      "items": "",
      "itemsPerPage": "Items per page",
      "optionsToggle": "Items per page",
      "page": "",
      "paginationTitle": "Pagination",
      "perPageSuffix": "per page",
      "toFirstPage": "Go to first page",
      "toLastPage": "Go to last page",
      "toNextPage": "Go to next page",
      "toPreviousPage": "Go to previous page",
    }
  }
  toggleTemplate={[Function]}
  variant="bottom"
  widgetId="pagination-options-menu"
/>
`;
